src/backend/commands/createas.c

```diff
@@ -1496,19 +1456,27 @@ check_ivm_restriction_walker(Node *node, check_ivm_restriction_context *ctx, int
 							 errmsg("aggregate function %s is not supported on incrementally maintainable materialized view", aggname)));
 				break;
 			}
+		case T_FromExpr:
+		case T_NullIfExpr: /* same as OpExpr */
+		case T_DistinctExpr: /* same as OpExpr */
+		case T_OpExpr:
+		case T_BoolExpr:
+		case T_CaseExpr:
 		case T_SubPlan:
 		case T_GroupingFunc:
 		case T_WindowFunc:
 		case T_FuncExpr:
-		case T_SQLValueFunction:
-		case T_Const:
-		case T_Param:
+		case T_CoerceViaIO:
+		case T_List:
+			{
+				expression_tree_walker(node, check_ivm_restriction_walker, ctx);
+				break;
+			}
 		default:
 			/* do nothing */
 			break;
 	}
-
```

Comment:
If I understand correctly, this function performs individual checks for some important nodes and just calls expression_tree_walker for other nodes. If so, do we need "case" for each other nodes? Can we call expression_tree_walker in "default" instead of writing many "case"s?

I am wondering that we have to maintain this case list when a new node is added in trees in future and  that it is possible to miss some cases.

Reply:
Thank you for your comments.
I left my fixes for clarity. I think it would be better to unify 'case' to 'default',
so I fix it.